src/amp-story-player/amp-story-entry-point/amp-story-entry-point-impl.js

```diff
@@ -84,6 +113,170 @@ export class AmpStoryEntryPoint {
+   * @param {string} src
+   * @param {string} styleClass
+   * @param {string} href
+   * @param {number} index
+   * @private
+   */
+  initializeCard_(cardContainer, src, styleClass, href, index) {
+    const card = this.doc_.createElement('img');
+    card.src = src;
+    card.classList.add(styleClass);
+    card.classList.add('entry-point-card-img');
+    this.setCardWidthHeight_(card, styleClass);
+    this.onCardClick_(card, href, index);
+    cardContainer.append(card);
+  }
+
+  /**
+   * Sets the correct height and width for poster based on style.
+   * @param {HTMLImageElement} card
+   * @param {string} styleClass
+   * @private
+   */
+  setCardWidthHeight_(card, styleClass) {
+    card.height = 233;
+    if (styleClass === 'circular-entry-point') {
+      card.width = 100;
+      card.height = 100;
+    }
+  }
+
+  /**
+   * Reacts to on click event for an entry point.
+   * @param {HTMLImageElement} card
+   * @param {string} href
+   * @param {number} index
+   * @private
+   */
+  onCardClick_(card, href, index) {
+    card.addEventListener('click', () => {
+      this.player_.show(href);
+      this.element_.dispatchEvent(
+        createCustomEvent(this.win_, 'entryPointClicked', {href, index})
+      );
+    });
+  }
+
+  /**
+   * Creates arrows to be displayed.
+   * @private
+   * */
+  initializeCarouselArrow_() {
+    const scrollContainer = this.rootEl_.querySelector('div.entry-points');
+    const cards = this.rootEl_.querySelectorAll(
+      'div.entry-point-card-container'
+    );
+    const maxScroll =
+      scrollContainer./*OK*/ clientWidth -
+      cards.length * cards[0]./*OK*/ clientWidth;
+    if (maxScroll >= 0) {
+      return;
+    }
+
+    this.createCarouselArrow_(true, maxScroll, cards);
+    this.createCarouselArrow_(false, maxScroll, cards);
+  }
+
+  /**
+   * Creates a button element, adds scrolling ability to it and displays it on the page.
+   * @param {boolean} isLeft
+   * @param {number} maxScroll
+   * @param {Array<Element>} cards
+   * @private
+   */
+  createCarouselArrow_(isLeft, maxScroll, cards) {
+    const button = this.doc_.createElement('button');
+    button.classList.add(
+      isLeft ? 'entry-point-left-arrow' : 'entry-point-right-arrow'
+    );
+    button.classList.add('entry-point-arrow');
+    this.rootEl_.appendChild(button);
+    this.addScrollingToArrow_(button, isLeft, maxScroll, cards);
+  }
+
+  /**
+   * Add ability to scroll to button.
+   * @param {Element} button
+   * @param {boolean} isLeft
+   * @param {number} maxScroll
+   * @param {Array<Element>} cards
+   * @private
+   */
+  addScrollingToArrow_(button, isLeft, maxScroll, cards) {
+    button.addEventListener('click', () => {
+      this.scrollX_ = isLeft
+        ? Math.min(0, this.scrollX_ + 50)
+        : Math.max(maxScroll, this.scrollX_ - 50);
+
+      for (let i = 0; i < cards.length; i++) {
+        setStyle(cards[i], 'transform', `translateX(${this.scrollX_}px)`);
```

Comment:
Another alternative would be to create a `style` tag, append it to before the buttons and update it with each `transform` value instead of adding the it to each card individually.